ui-v2/app/components/tabular-collection.js

```diff
@@ -106,6 +143,108 @@ export default Component.extend(SlotsMixin, {
       needsRevalidate(this);
     }
   },
+  // need to overwrite this completely so I can pass through the checked index
+  // for `formatItemStyle` in 3 places
+  updateCells: function() {
+    if (!this._items) {
+      return;
+    }
+    const numItems = get(this._items, 'length');
+    if (this._cellLayout.length !== numItems) {
+      this._cellLayout.length = numItems;
+    }
+
+    var priorMap = this._cellMap;
+    var cellMap = Object.create(null);
+
+    var index = this._cellLayout.indexAt(
+      this._scrollLeft,
+      this._scrollTop,
+      this._clientWidth,
+      this._clientHeight
+    );
+    var count = this._cellLayout.count(
+      this._scrollLeft,
+      this._scrollTop,
+      this._clientWidth,
+      this._clientHeight
+    );
+    var items = this._items;
+    var bufferBefore = Math.min(index, this._buffer);
+    index -= bufferBefore;
+    count += bufferBefore;
+    count = Math.min(count + this._buffer, get(items, 'length') - index);
+    var i, style, itemIndex, itemKey, cell;
+
+    var newItems = [];
+
+    for (i = 0; i < count; i++) {
+      itemIndex = index + i;
+      itemKey = identity(items.objectAt(itemIndex));
+      if (priorMap) {
+        cell = priorMap[itemKey];
+      }
+      if (cell) {
+        // additional `checked` argument
+        style = this._cellLayout.formatItemStyle(
+          itemIndex,
+          this._clientWidth,
+          this._clientHeight,
+          this.checked
```

Comment:
is `checked` the "You've clicked the menu on this item" ? state?

Reply:
Ah no, I see what it looks like :) but it's not a boolean, it's the index of the checked 'cell'. I think this was the 'best' way I could think of to fiddle with as little of `ember-collection` as possible.

I already have a custom `ZIndexedGrid` up above, so I'm passing the 'checked index' through here in 3 places, and I've done the logic up in in one place in my already custom `ZIndexedGrid`. I commented up the 3 places where I had to change the `ember-collection` code by passing through an extra arg also.

Reply:
Ahh I thought `checked` like the form element state, not an index.